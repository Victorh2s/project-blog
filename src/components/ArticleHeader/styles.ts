import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`

  text-align: center;
  justify-content: center;
  margin-bottom: ${theme.spacings.xlarge};
  border-bottom: 0.1rem solid ${theme.colors.darkeyGray};

  ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }

  @media ${theme.media.lteMedium} {
      ${HeadingStyles}{
      font-size: ${theme.font.sizes.large};
    }
  }

`}
`;

export const Text = styled.p`
  text-align: center;
  justify-content: center;
`;

export const Img = styled.img`
  max-width: 100%;
  display: block;
`;
