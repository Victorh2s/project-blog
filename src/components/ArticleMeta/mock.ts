import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-06-10T15:16:09.246+00:00',
  author: {
    id: '123456789',
    slug: 'victor-henrique',
    displayName: 'Victor H',
  },
  categories: [
    {
      id: 'asdslgyh65',
      displayName: 'Tech',
      slug: 'tech',
    },
    {
      id: 'asdslgyh66',
      displayName: 'JS',
      slug: 'javascript',
    },
  ],
} as ArticleMetaProps;
