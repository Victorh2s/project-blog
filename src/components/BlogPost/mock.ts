import { BlogPostProps } from '.';
import { data } from '../api/dados.json';

const { title, excerpt, content, cover, author, categories, createdAt } =
  data.posts[0];

export default {
  title,
  excerpt,
  content,
  cover,
  author,
  categories,
  createdAt,
} as BlogPostProps;
