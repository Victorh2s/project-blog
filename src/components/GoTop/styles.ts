import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    color: ${theme.colors.primary};
    border: 0.1rem solid;
    border-radius: 50rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
  `}
`;
