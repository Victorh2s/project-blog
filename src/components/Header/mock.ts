import { HeaderProps } from '.';

export default {
  BlogName: 'Usu√°rio teste',
  BlogDescription:
    'Et sed legere rationibus, vitae evertitur ad eam. Eu amet vero officiis vis. Ad nullam pericula neglegentur sit, vix ei regione legendos argumentum. Mei an quas convenire, nobis dicunt mei ut.',
  logo: 'assets/images/logo.png',
  showText: true,
} as HeaderProps;
