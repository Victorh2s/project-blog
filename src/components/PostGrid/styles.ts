import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  width: 100%;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
  max-width: 120rem;
  max-height: 300rem;
  box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
  gap: ${theme.spacings.large};
  padding: ${theme.spacings.large};
  text-overflow: ellipsis;

  @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const NotFound = styled.p`
${({ theme }) => css`

    padding: ${theme.spacings.large};
    text-align: center;
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.secondary};

`}
`;
