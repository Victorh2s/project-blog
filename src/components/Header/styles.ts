import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';
import { Container as LogoLinkstyles } from '../LogoLink/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacings.huge};
    width: 100%;
    max-width: 96rem;
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }

    ${LogoLinkstyles}{
      >img{
        border: 0.3rem solid;
        border-color:  rgb(74 222 128);
      }
    }

    @media ${theme.media.lteMedium}{
      display:flex;
      flex-flow: column wrap;

      & > ${HeadingStyles} {
        margin: 0 0 ${theme.spacings.medium} 0;
      }

    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;

    @media ${theme.media.lteMedium}{
      margin-left: 0;
    }
  `}
`;
