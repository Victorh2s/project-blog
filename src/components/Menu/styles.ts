import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  Visible: boolean;
};

const ContainerChanger = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
  left: ${Visible ? 0 : '-35rem'};
  overflow-y: ${Visible ? 'auto' : 'hidden'};

  @media ${theme.media.lteMedium} {
    left: ${Visible ? '0' : '-35rem'};
  }
`;

export const Container = styled.div<MenuBehaviorProps>`
  ${({ theme, Visible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;
    ${ContainerChanger(Visible, theme)}
  `}
`;

export const Nav = styled.nav`
    margin: 0 auto;
    width: 100%;
`;

export const MenuLinks = styled.div`

   margin-bottom: 5rem;

`;

export const Logo = styled.div`

    ${HeadingStyles} {

      img{
        margin: 0 auto;
        margin-bottom: 3rem;
      }
    }

`;

const ButtonChanger = (
  Visible: MenuBehaviorProps['Visible'],
  theme: DefaultTheme,
) => css`
  left: ${Visible ? '26rem,' : '1rem'};
  color: ${Visible ? theme.colors.secondary : theme.colors.primary};

  @media ${theme.media.lteMedium} {
    left: ${Visible ? '26rem' : '-0.5rem'};
  }
`;

export const Button = styled.a<MenuBehaviorProps>`
  ${({ theme, Visible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    left: 26rem;
    color: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    transition: all 300ms ease-in-out;
    ${ButtonChanger(Visible, theme)}
  `}
`;
