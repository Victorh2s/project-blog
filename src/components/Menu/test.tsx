import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Menu, MenuProps } from '.';
import mock from './mock';

const props: MenuProps = mock;

describe('<Menu />', () => {
  it('should render button link', () => {
    renderTheme(<Menu {...props} links={undefined} />);

    const button = screen.getByRole('link', { name: 'Open or Close Menu' });
    const openMenuIcon = screen.getByLabelText('Open Menu');

    expect(button).toBeInTheDocument();
    expect(openMenuIcon).toBeInTheDocument();

    expect(screen.queryByLabelText('Close Menu')).not.toBeInTheDocument();
  });

  it('should open/close menu on button click', () => {
    const { container } = renderTheme(<Menu {...props} />);

    const button = screen.getByRole('link', { name: 'Open or Close Menu' });
    fireEvent.click(button);

    expect(screen.queryByLabelText('Close Menu')).toBeInTheDocument();
    expect(screen.queryByRole('navigation')).toBeInTheDocument();
    expect(screen.queryByLabelText('Open Menu')).not.toBeInTheDocument();
    expect(screen.getByRole('heading', { name: 'Victor' }));
    expect(screen.getByRole('img', { name: 'Victor' }));
    expect(
      screen.getByRole('navigation').querySelectorAll('a:not([href="/"])'),
    ).toHaveLength(mock.links.length);

    fireEvent.click(button);

    expect(screen.queryByLabelText('Close Menu')).not.toBeInTheDocument();
    expect(screen.queryByLabelText('Open Menu')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
