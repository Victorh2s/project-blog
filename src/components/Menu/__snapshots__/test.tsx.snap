// Vitest Snapshot v1

exports[`<Menu /> > should open/close menu on button click 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
  background: #333333;
}

.c6 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c9 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #0A1128;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c9:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c2 {
  background: #0A1128;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -35rem;
  overflow-y: hidden;
}

.c3 {
  margin: 0 auto;
  width: 100%;
}

.c8 {
  margin-bottom: 5rem;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c4 .c5 img {
  margin: 0 auto;
  margin-bottom: 2rem;
  border: 0.1rem solid;
  border-color: #dc143c;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  left: 26rem;
  color: #0A1128;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: 1rem;
  color: #0A1128;
}

@media (max-width:768px) {
  .c2 {
    left: -35rem;
  }
}

@media (max-width:768px) {

}

@media (max-width:768px) {
  .c0 {
    left: -0.5rem;
  }
}

@media (max-width:768px) {

}

<div>
  <a
    aria-label="Open or Close Menu"
    class="c0"
    href="#"
    title="Open or Close Menu"
  >
    <svg
      aria-hidden="true"
      aria-label="Open Menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </a>
  <div
    aria-hidden="true"
    class="c2"
  >
    <nav
      class="c3"
    >
      <div
        class="c4"
      >
        <h1
          class="c5 c6"
        >
          <a
            class="c7"
            href="/"
            target="_self"
          >
            <img
              alt="Victor"
              src="assets/images/logo.png"
            />
          </a>
        </h1>
      </div>
      <div
        class="c8"
      >
        <a
          class="c9"
          href="/home/1"
          target="_self"
        >
          My jobs
        </a>
        <a
          class="c9"
          href="/home/2"
          target="_self"
        >
          About me
        </a>
        <a
          class="c9"
          href="/home/3"
          target="_self"
        >
          Services
        </a>
        <a
          class="c9"
          href="http://localhost/home/4"
          target="_self"
        >
          External Link
        </a>
        <a
          class="c9"
          href="/home/5"
          target="_self"
        >
          JavaScript
        </a>
        <a
          class="c9"
          href="/home/6"
          target="_self"
        >
          Python
        </a>
        <a
          class="c9"
          href="/home/7"
          target="_self"
        >
          Consectetur
        </a>
        <a
          class="c9"
          href="/home/8"
          target="_self"
        >
          Text test
        </a>
        <a
          class="c9"
          href="/home/9"
          target="_self"
        >
          Pac-man
        </a>
        <a
          class="c9"
          href="/home/10"
          target="_self"
        >
          Star Wars
        </a>
        <a
          class="c9"
          href="/home/11"
          target="_self"
        >
          A menu
        </a>
      </div>
    </nav>
  </div>
</div>
`;
